### PR TITLE
[Python] Add enum support when building default values for model properties when using `$ref`

### PR DESCRIPTION
**Pull request by welshm**
Add in better processing for enums in Python when they are a $ref

- Currently, doesn't properly populate default values so when models are de/serialized, the enum value is dropped.
- Adds support for checking the $ref when getting the default value for a Schema
- Updates how enum variable names are referenced
  - Update `DefaultCodegen.java` with `public String toEnumDefaultValue(CodegenProperty property, String value)`
  - Allows checking if the property is an enum reference - and if so, handle it as a first class enum model in Python
  - Otherwise, use the inline variable value (as it does today)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -6750,7 +6762,7 @@ public void updateCodegenPropertyEnum(CodegenProperty var) {
                 }
             }
             if (enumName != null) {
-                var.defaultValue = toEnumDefaultValue(enumName, var.datatypeWithEnum);
+                var.defaultValue = toEnumDefaultValue(var, enumName);
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:6765]: fyi @OpenAPITools/generator-core-team

File: bin/configs/python-flask.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 generatorName: python-flask
-outputDir: samples/server/petstore/python-flask
+outputDir: samples/server/petstore/2_0/python-flask
 inputSpec: modules/openapi-generator/src/test/resources/2_0/python-flask/petstore.yaml
 templateDir: modules/openapi-generator/src/main/resources/python-flask
```
Comment (wing328) [inline bin/configs/python-flask.yaml:4]: we don't test 2.0 spec so no need to keep this sample folder and yaml config file

**Comment by welshm** [comment]
Something went wrong doing the rebase - will fix

**Comment by wing328** [comment]
thanks for the PR. the commit history looks ok as we will squash the commits during merge.

tested python client with the new test schemas and the result is good: https://github.com/OpenAPITools/openapi-generator/pull/18882

**Comment by wing328** [comment]
fyi. filed https://github.com/OpenAPITools/openapi-generator/pull/18883 to add the tests in python client as well

**Comment by wing328** [comment]
@welshm when you've time, can you please PM me via Slack? thanks.

**Pull request merged**